pages/examples/web3-authentication.tsx

```diff
@@ -0,0 +1,207 @@
+              {JSON.stringify(currentUser, null, 2)}
+            </MonospacePreview>
+            <div style={{ marginTop: 16 }}>
+              <ActionItem
+                icon={`✳`}
+                onClick={async () => {
+                  const response = await Queries.userAuthenticate({ email, password });
+                  if (!response) {
+                    setModal({
+                      name: 'ERROR',
+                      message: 'Something went wrong. This is also a lazy message. Ideally the error message would have told you that you forgot to put your email or password.',
+                    });
+                    return;
+                  }
+                  setUser(response.user);
+                  alert('The user data was refreshed.');
+                }}
+              >
+                Refresh
+              </ActionItem>
+              <ActionItem
+                icon={`✳`}
+                onClick={async () => {
+                  const response = await Queries.userRefreshKey({ email, password });
+                  if (!response) {
+                    setModal({
+                      name: 'ERROR',
+                      message: 'Something went wrong. This is also a lazy message. Ideally the error message would have told you that you forgot to put your email or password.',
+                    });
+                    return;
+                  }
+                  Cookies.remove('sitekey');
+                  setUser(response.user);
+                  alert('A new API key was issued. You will need to set the Cookie again.');
+                }}
+              >
+                Generate new API key
+              </ActionItem>
+              <ActionItem
+                icon={`✳`}
+                onClick={async () => {
+                  setKey(currentUser.key);
+                  Cookies.set('sitekey', currentUser.key, { domain: props.host, secure: true });
+                  alert('Your API key was attached to a cookie with domain and secure options.');
+                }}
+              >
+                Set Cookie (persistent session / sign in)
+              </ActionItem>
+              <ActionItem
+                icon={`⭡`}
+                onClick={() => {
+                  setUser(null);
+                }}
+              >
+                Hide user information
+              </ActionItem>
+              <ActionItem icon={`⭢`} href="/examples/settings">
+                View settings
+              </ActionItem>
+            </div>
+          </>
+        ) : null}
+      </ThinAppLayout>
+      <GlobalModalManager currentModal={currentModal} setModal={setModal} onHandleThemeChange={Utilities.onHandleThemeChange} />
+    </Page>
+  );
+}
+
+export async function getServerSideProps(context) {
+  let viewer = null;
+  let sessionKey = context.req.cookies['sitekey'] || '';
+  if (Utilities.isEmpty(sessionKey)) {
+    return {
+      props: { sessionKey: '', viewer: null },
```

Comment:
From `import * as Constants from "@common/constants"`

then add `export const WHATEVER_YOU_WANT_TO_CALL_IT = 8f9365e47361eb8a18bb52144e775141;`

then return it as a prop here.

and then the component access it from `props.whateverYouCalledIt`

and this would support people who want to obfuscate this value from the server. (if they wanted to)